tests/5.1/strict/test_strict_grainsize.c

```diff
@@ -0,0 +1,48 @@
+//===--- test_strict_grainsize.c -----------------------------------------------===//
+//
+//  OpenMP API Version 5.1 Aug 2021
+//
+// This test checks the behavior of taskloop's clause grainsize, when the strict 
+// modifier is present. The grainsize strict expressions specifies the exact number of 
+// logical iterations which should be performed per task, except for the sequentially
+// last iteration which may have fewer than the specified iterations. 
+////===--------------------------------------------------------------------------------------===//
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+#include <math.h>
+
+#define N 1024
+
+int errors;
+
+int test_strict_grainsize() {
+  int arr[N];
+  int sum = 0;
+  int parallel_sum = 0; 
+  for (int i=0; i<N; i++){
+        arr[i] = 1;
+        sum += arr[i];
+ }
+#pragma omp parallel  
+#pragma omp single
+#pragma omp taskloop grainsize(strict:1000)
+  for (int i = 0; i < N; i++) {
+  	parallel_sum += arr[i];
```

Comment:
This works with GCC 12 – but only if I add `reduction(+:parallel_sum)`

Reply:
Good catch - thanks! Updated.